pkg/eventfilter/subscriptionsapi/all_filter.go

```diff
@@ -65,7 +65,7 @@ func (filter *allFilter) Filter(ctx context.Context, event cloudevents.Event) ev
 	filter.rwMutex.RLock()
 	defer filter.rwMutex.RUnlock()
 	for i, f := range filter.filters {
-		res = res.And(f.filter.Filter(ctx, event))
+		res = f.filter.Filter(ctx, event)
```

Comment:
A question: why can we remove the And outside?

Reply:
Because we are short circuiting the result whenever it fails. If it doesn't fail, then res will be successful at the end of the loop. Note that all the unit tests still pass after removing this logic :)

Reply:
that makes a lot of sense!